Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE_HeightFuse.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using RimWorld;
+using UnityEngine;
+using Verse;
+using Verse.Sound;
+using CombatExtended.Compatibility;
+using CombatExtended.Lasers;
+using ProjectileImpactFX;
+using CombatExtended.Utilities;
+
+namespace CombatExtended
+{
+    class ProjectileCE_HeightFuse : ProjectileCE
+    {
+        float detonationHeight => (def.projectile as ProjectilePropertiesCE).aimHeightOffset;
+
+        bool armed;
+
+        public override void ExposeData()
+        {
+            base.ExposeData();
+            Scribe_Values.Look(ref armed, "armed", false);
+        }
+
+        public override void Tick()
+        {
+            base.Tick();
+            if (!armed && LastPos.y > detonationHeight)
+            {
+                armed = true;
+            }
+            if (armed && Height <= detonationHeight)
+            {
+                HeightFuseAirBurst();
+            }
+        }
+
+        public override void Impact(Thing hitThing)
+        {
+            //intercept impact if it hit something after where height fuse should have triggered
+            if (armed && Height <= detonationHeight)
```

Comment:
Not required.  If LastPos wasn't high enough, it won't be armed.  If it *was* high enough, and then fell too low *last* tick, then it will have detonated mid-air last tick.  To be at this tick, armed, and lower than detonation height, last tick it must have been above detonation height.  We could add an else-throw in debug builds if you want to be certain no one in the future breaks it, but I don't think it's needed.